**В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что:** 
 
            a) все мячи белые?;  
            б) ровно два мяча белые?;  
            в) хотя бы один мяч белый?.

Для решения данной задачи подсчитаем общее число элементарных исходов при вытаскивании мячей:

$n_1=C^2_{10}$ - для первой корзины  
$n_2=C^2_{11}$ - для второй корзины  
Напишем функцию, для подсчёта количества комбинаций:

In [1]:
from math import factorial
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

Вычислим $n_1$ и $n_2$:

In [2]:
n1=combinations(10,2)
n2=combinations(11,2)
print(f'Общее число элементарных исходов для первой корзины: n1 = {n1}\n'
f'Общее число элементарных исходов для второй корзины: n2 = {n2}')

Общее число элементарных исходов для первой корзины: n1 = 45
Общее число элементарных исходов для второй корзины: n2 = 55


а) Для выполнения условия, что вытащены только белые мячи, необходимо одновременное наступление двух событий:  
событие A-из первой корзины извлекли 2 белых мяча,  
событие B-из второй корзины извлекли 2 белых мяча.  
Для события A число благоприятных элементарных исходов $m_1=C^2_{7}$  
Для события B число благоприятных элементарных исходов $m_2=C^2_{9}$


Вычислим $m_1$ и $m_2$:

In [3]:
m1=combinations(7,2)
m2=combinations(9,2)
print(f'Число благоприятных элементарных исходов для события А: m1 = {combinations(7,2)}\n'
      f'Число благоприятных элементарных исходов для события В: m2 = {combinations(9,2)}'
     )

Число благоприятных элементарных исходов для события А: m1 = 21
Число благоприятных элементарных исходов для события В: m2 = 36


Для событий A и B вероятности их наступления будут расчитываться как $P(A)=\frac{m_1}{n_1}$
и $P(B)=\frac{m_2}{n_2}$, а вероятность их одновременного наступления как $P(A+B)=P(A)\cdot P(B)$

Расчитаем:

In [4]:
print(f'Вероятность наступления события А, Р(А) = {m1/n1: .4f}\n'
      f'Вероятность наступления события В, Р(В) = {m2/n2: .4f}\n'
      f'Вероятность одновременного наступления событий А и В, Р(А+В) = {(m1/n1)*(m2/n2): .4f}'
     )

Вероятность наступления события А, Р(А) =  0.4667
Вероятность наступления события В, Р(В) =  0.6545
Вероятность одновременного наступления событий А и В, Р(А+В) =  0.3055


Решим через условную вероятность:  
$P(A+B)=\frac{7}{10}\cdot\frac{6}{9}\cdot\frac{9}{11}\cdot\frac{8}{10}$

In [5]:


P=(7/10)*(6/9)*(9/11)*(8/10)
print(f'Вероятность одновременного наступления событий А и В, Р(А+В) = {P: .4f}')

Вероятность одновременного наступления событий А и В, Р(А+В) =  0.3055


___  
_Ответ: вероятность того, что все вытащенные мячи белые составляет **P=0.3055** или **$\approx$ 30.55%**_

___

б) событию, в котором вытащили ровно два белых мяча, соответствует несколько комбинаций:  
1. События A-из первой корзины извлекли 2 белых мяча и B-из второй корзины извлекли 2 мяча другого цвета, **или**  
2. События C-из первой корзины извлекли 1 белый мяч и 1 мяч другого цвета и D-из второй корзины извлекли 1 белый мяч и 1 мяч другого цвета, **или**  
3. События E-из первой корзины извлекли 2 мяч другого цвета и F-из второй корзины извлекли 2 белого цвета.  

А вероятность того, что сложится одна из этих комбинаций, будет суммой вероятностей этих комбинаций.  
Т.е.: $P=P(A)\cdot P(B)+P(C)\cdot P(D)+P(E)\cdot P(F)$   

Число благоприятных элементарных исходов для каждого события в комбинации:
1. $m_A=C^2_{7}$,$m_B=C^0_{9}$  
2. $m_C=C^1_{7}$,$m_D=C^1_{9}$  
3. $m_E=C^0_{7}$,$m_F=C^2_{9}$  
$\Rightarrow P=\frac{m_A}{n_1}\cdot\frac{m_B}{n_2}+\frac{m_C}{n_1}\cdot\frac{m_D}{n_2}+\frac{m_E}{n_1}\cdot\frac{m_F}{n_2}=\frac{C^2_{7}}{n_1}\cdot\frac{C^0_{9}}{n_2}+\frac{C^1_{7}\cdot C^1_3}{n_1}\cdot\frac{C^1_{9}\cdot C^1_2}{n_2}+\frac{C^0_{7}}{n_1}\cdot\frac{C^2_{9}}{n_2}$


Произведем вычисления:

In [6]:
m_A=combinations(7,2)
m_B=combinations(2,2)
m_C=combinations(7,1)*combinations(3,1)
m_D=combinations(9,1)*combinations(2,1)
m_E=combinations(3,2)
m_F=combinations(9,2)
print(f'Число благоприятных элементарных исходов для события А: m_A = {m_A}\n'
      f'Число благоприятных элементарных исходов для события B: m_B = {m_B}\n'
      f'Число благоприятных элементарных исходов для события C: m_C = {m_C}\n'
      f'Число благоприятных элементарных исходов для события D: m_D = {m_D}\n'
      f'Число благоприятных элементарных исходов для события E: m_E = {m_E}\n'
      f'Число благоприятных элементарных исходов для события F: m_F = {m_F}'  
     )

Число благоприятных элементарных исходов для события А: m_A = 21
Число благоприятных элементарных исходов для события B: m_B = 1
Число благоприятных элементарных исходов для события C: m_C = 21
Число благоприятных элементарных исходов для события D: m_D = 18
Число благоприятных элементарных исходов для события E: m_E = 3
Число благоприятных элементарных исходов для события F: m_F = 36


In [7]:

P=(m_A/n1)*(m_B/n2)+(m_C/n1)*(m_D/n2)+(m_E/n1)*(m_F/n2)
print(f'Вероятность того, что будут вытащены ровно 2 белых мяча Р = {P: .4f}')

Вероятность того, что будут вытащены ровно 2 белых мяча Р =  0.2048


Решим через условную вероятность:  
$P=\frac{7}{10}\cdot\frac{6}{9}\cdot\frac{2}{11}\cdot\frac{1}{10}+4^*\cdot\left(\frac{7}{10}\cdot\frac{3}{9}\cdot\frac{9}{11}\cdot\frac{2}{10}\right)+\frac{3}{10}\cdot\frac{2}{9}\cdot\frac{9}{11}\cdot\frac{8}{10}$

_Примечание. Второе слагаемое нужно умножить на 4, т.к. существует 4 способа достать из каждой корзины 1 белый мяч и 1 мяч другого цвета._

In [8]:
P1=((7/10)*(6/9)*(2/11)*(1/10))+4*((7/10)*(3/9)*(9/11)*(2/10))+((3/10)*(2/9)*(9/11)*(8/10))
print(f'Вероятность того, что будут вытащены ровно 2 белых мяча Р = {P1: .4f}')

Вероятность того, что будут вытащены ровно 2 белых мяча Р =  0.2048


___  
_Ответ: вероятность того, что будут вытащены ровно 2 белых мяча составляет **P=0.2048** или **$\approx$ 20.48%**_

___

в) В данном случае удобнее рассмотреть событие $\bar A$ -"не было извлечено ни одного белого мяча", которое наряду с искомым событием $A$-"был извлечен хотя бы 1 белый мяч", составляют полную группу событий, т.к. в результате испытаний может произойти только либо событие $A$, либо событие $\bar A$, и поэтому:  
$P(A)=1-P(\bar A)$  
Для наступления события $\bar A$ необходимо чтобы из первой и второй корзин не было вынуто ни одного белого мяча.  
Тогда число таких благоприятных исходов:  
для первой корзины - $m_1=C^2_{3}$  
для второй корзины -  $m_2=C^2_{2}$


Вычислим $m_1$ и $m_2$:

In [9]:
m1=combinations(3,2)
m2=combinations(2,2)
print(f'Число благоприятных элементарных исходов для первой корзины: m1 = {combinations(3,2)}\n'
      f'Число благоприятных элементарных исходов для второй корзины: m2 = {combinations(2,2)}'
     )

Число благоприятных элементарных исходов для первой корзины: m1 = 3
Число благоприятных элементарных исходов для второй корзины: m2 = 1


Тогда $P(\bar A)=\frac{m1}{n1}\cdot\frac{m2}{n2}$, вычислим $P(\bar A)$ и $P(A)$

In [10]:
print(f'Вероятность того, что не будет извлечено ни одного белого мяча Р(А_обр) = {(m1/n1)*(m2/n2): .4f}\n'
      f'Вероятность того, что будет извлечен хотя бы один белый мяч Р(А) = 1 - Р(А_обр) = {1-(m1/n1)*(m2/n2): .4f}'
     )

Вероятность того, что не будет извлечено ни одного белого мяча Р(А_обр) =  0.0012
Вероятность того, что будет извлечен хотя бы один белый мяч Р(А) = 1 - Р(А_обр) =  0.9988


Решим через условную вероятность:  
$P(A)=1-\frac{3}{10}\cdot\frac{2}{9}\cdot\frac{2}{11}\cdot\frac{1}{10}$

In [11]:
P=1-(3/10)*(2/9)*(2/11)*(1/10)
print(f'Вероятность того, что будет извлечен хотя бы один белый мяч Р(А) = {P: .4f}')

Вероятность того, что будет извлечен хотя бы один белый мяч Р(А) =  0.9988


___  
_Ответ: вероятность того, что будет вытащен хотя бы 1 белый мяч составляет **P=0.9988** или **$\approx$ 99.88%**_

___